cmd/es-node/main.go

```diff
@@ -235,13 +235,14 @@ func EsNodeInit(ctx *cli.Context) error {
 		for i := 0; i < len(shardIndexes); i++ {
 			shard := uint64(shardIndexes[i])
 			if shard > 0 {
-				diff, err := getDifficulty(cctx, client, l1Contract, shard)
+				timeStamp, err := getLastMineTime(cctx, client, l1Contract, shard)
 				if err != nil {
 					log.Error("Failed to get shard info from contract", "error", err)
 					return err
 				}
-				if diff != nil && diff.Cmp(big.NewInt(0)) == 0 {
-					return fmt.Errorf("Shard not exist: %d", shard)
+				// lastMineTime will be set to non-zero when opening a new shard
+				if timeStamp != nil && timeStamp.Cmp(big.NewInt(0)) == 0 {
```

Comment:
This means people cannot create a new shard file and pre-fill it with empty entries before the contract opens a new shard.